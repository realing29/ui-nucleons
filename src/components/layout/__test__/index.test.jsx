import React from 'react';
import { mount } from 'enzyme';
import { DesktopLayout, MobileLayout } from '..';

describe('Layouts', () => {
  [DesktopLayout, MobileLayout].forEach(Layout => {
    it('should renders correctly', () => {
      const wrapper = mount(
        <Layout className='test-class' style={{ height: 128 }}>
          Hello, world!
        </Layout>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should handle "element" prop', () => {
      const wrapper = mount(
        <Layout element='main' className='test-class' style={{ height: 128 }}>
          Hello, world!
        </Layout>
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
