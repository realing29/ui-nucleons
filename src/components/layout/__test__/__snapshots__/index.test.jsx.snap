// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts should handle "element" prop 1`] = `
<DesktopLayout
  className="test-class"
  element="main"
  style={
    Object {
      "height": 128,
    }
  }
>
  <main
    className="layout desktop test-class"
    style={
      Object {
        "height": 128,
      }
    }
  >
    Hello, world!
  </main>
</DesktopLayout>
`;

exports[`Layouts should handle "element" prop 2`] = `
<MobileLayout
  className="test-class"
  element="main"
  style={
    Object {
      "height": 128,
    }
  }
>
  <main
    className="layout mobile test-class"
    style={
      Object {
        "height": 128,
      }
    }
  >
    Hello, world!
  </main>
</MobileLayout>
`;

exports[`Layouts should renders correctly 1`] = `
<DesktopLayout
  className="test-class"
  style={
    Object {
      "height": 128,
    }
  }
>
  <div
    className="layout desktop test-class"
    style={
      Object {
        "height": 128,
      }
    }
  >
    Hello, world!
  </div>
</DesktopLayout>
`;

exports[`Layouts should renders correctly 2`] = `
<MobileLayout
  className="test-class"
  style={
    Object {
      "height": 128,
    }
  }
>
  <div
    className="layout mobile test-class"
    style={
      Object {
        "height": 128,
      }
    }
  >
    Hello, world!
  </div>
</MobileLayout>
`;
